### DISCUSSION TITLE
How can I sort Datagrid under AOT?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#14422] [Q&A]
I know Datagrid doesn't work very well under AOT, but this is the only functionality I need. I copied the code from [https://docs.avaloniaui.net/docs/reference/controls/datagrid/](https://docs.avaloniaui.net/docs/reference/controls/datagrid/), but sorting doesn't seem to work.

**Comment by maxkatz6** [comment]
There is a known issue that primitive types cannot be sorted in AOT, like int/float. But it should be fine with reference types.

**Reply by IronEaterBeast** [reply]
It seems that DynamicDependency still needs to be specified in order to sort, and the method described in the documentation does not work directly. For example:

[DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(TestModel))]
public readonly ObservableCollection<TestModel> Persons = new();

**Reply by maxkatz6** [reply]
You should use CompiledBinding on DataGrid as well. You wouldn't need DynamicDependency in this case.

**Reply by IronEaterBeast** [reply]
How to sort the contents of a DataGridTemplateColumn, for example:

![image](https://github.com/AvaloniaUI/Avalonia/assets/9688223/f523114d-d197-4c1b-9281-b5f60dc18f80)

I placed a URL in this column, clicking it will open the link.

**Reply by maxkatz6** [reply]
DataGridTemplateColumn is not sortable. There is no information to sort.
If you need to, you can set SortMemberPath property on DataGridTemplateColumn. But this one **might be** trimmed by AOT without any extra attributes.

**Reply by IronEaterBeast** [reply]
Is it possible to specify a default sorting column, such as automatically sorting by the 'time' DataGridTextColumn when displaying the DataGrid, instead of the order of the data itself?
